commons/src/main/java/org/dhis2/commons/dialogs/imagedetail/ImageDetailActivity.kt

```diff
@@ -0,0 +1,79 @@
+package org.dhis2.commons.dialogs.imagedetail
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.core.content.FileProvider
+import org.dhis2.commons.R
+import org.dhis2.commons.data.FormFileProvider
+import org.dhis2.commons.extensions.getBitmap
+import org.hisp.dhis.mobile.ui.designsystem.component.FullScreenImage
+import timber.log.Timber
+import java.io.File
+import java.io.IOException
+
+class ImageDetailActivity : AppCompatActivity() {
+
+    companion object {
+
+        private const val ARG_IMAGE_TITLE = "arg_image_title"
+        private const val ARG_IMAGE_PATH = "arg_image_path"
+
+        fun intent(context: Context, title: String?, imagePath: String): Intent {
+            return Intent(context, ImageDetailActivity::class.java).apply {
+                putExtra(ARG_IMAGE_TITLE, title)
+                putExtra(ARG_IMAGE_PATH, imagePath)
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val title = intent.getStringExtra(ARG_IMAGE_TITLE)
+        val imagePath = intent.getStringExtra(ARG_IMAGE_PATH)!!
+
+        setContent {
+            val painter = remember(imagePath) {
+                imagePath.getBitmap()?.let { BitmapPainter(it.asImageBitmap()) }
+            }
+
+            FullScreenImage(
+                painter = painter!!,
+                title = title.orEmpty(),
+                onDismiss = { finish() },
+                onDownloadButtonClick = { saveImage(imagePath) },
+                onShareButtonClick = { shareImage(imagePath) },
+            )
+        }
+    }
+
+    private fun saveImage(image: String) {
+        // TODO: Implement saving image to disk
```

Comment:
I would develop a new download image implementation given that the one in Qr it seem is not working properly, and then this new implementation could be reused in other components like QR one. What do you think?

Reply:
Yeah, we would have to save the image publicly accessible folder like downloads. That should make the implementation simpler. We can do this in another story?

We also need to replace it in other places where the `InputImage` component is used as well (in Forms?), so we can do replace all the save image functionality in a single story. WDYT?

Reply:
yes, it is ok. so we could merge this without implementation and the develop the other one separately